src/vaadin-menu-bar-interactions-mixin.html

```diff
@@ -0,0 +1,342 @@
+          }
+        } else {
+          this._navigateByKey(event);
+        }
+      }
+    }
+
+    _navigateByKey(event) {
+      // IE names for arrows do not include the Arrow prefix
+      const key = event.key.replace(/^Arrow/, '');
+      const buttons = this._buttons;
+      const currentBtn = this.shadowRoot.activeElement || this._expandedButton;
+      const currentIdx = buttons.indexOf(currentBtn);
+      let idx;
+      let increment;
+
+      switch (key) {
+        case 'Left':
+          increment = -1;
+          idx = currentIdx - 1;
+          break;
+        case 'Right':
+          increment = 1;
+          idx = currentIdx + 1;
+          break;
+        case 'Home':
+          increment = 1;
+          idx = 0;
+          break;
+        case 'End':
+          increment = -1;
+          idx = buttons.length - 1;
+          break;
+        default:
+          // do nothing.
+      }
+
+      idx = this._getAvailableIndex(idx, increment, buttons);
+      if (idx >= 0) {
+        event.preventDefault();
+        const btn = buttons[idx];
+        const wasExpanded = currentBtn === this._expandedButton;
+        if (wasExpanded) {
+          this._close();
+        }
+        this._focusButton(btn);
+        if (wasExpanded && btn.item && btn.item.children) {
+          this.__openSubMenu(btn, {keepFocus: true});
+        }
+      }
+    }
+
+    _getAvailableIndex(index, increment, buttons) {
+      const totalItems = buttons.length;
+      let idx = index;
+      for (let i = 0; typeof idx === 'number' && i < totalItems; i++, idx += increment || 1) {
+        if (idx < 0) {
+          idx = totalItems - 1;
+        } else if (idx >= totalItems) {
+          idx = 0;
+        }
+
+        const btn = buttons[idx];
+        if (!btn.disabled && !btn.hasAttribute('hidden')) {
+          return idx;
+        }
+      }
+      return -1;
+    }
+
+    get _subMenu() {
+      return this.shadowRoot.querySelector('vaadin-context-menu');
+    }
+
+    __alignOverlayPosition(e) {
+      const overlay = e.target;
+      const getRect = () => this._expandedButton.getBoundingClientRect();
+      if (overlay.hasAttribute('bottom-aligned')) {
+        overlay.style.bottom = parseInt(getComputedStyle(overlay).bottom) + getRect().height + 'px';
+      }
+      if (overlay.hasAttribute('right-aligned')) {
+        overlay.style.right = parseInt(getComputedStyle(overlay).right) - getRect().width + 'px';
+      }
+    }
+
+    _itemsChanged(items, splices) {
+      const subMenu = this._subMenu;
+      if (subMenu && subMenu.opened) {
+        subMenu.close();
+      }
+    }
+
+    _onMouseOver(e) {
+      const button = Array.from(e.composedPath()).filter(el => el.item)[0];
+      if (button && button !== this._expandedButton && (this.openOnHover || (button.item.children && this._subMenu.opened))) {
+        this.__openSubMenu(button);
+      }
+    }
+
+    __onContextMenuKeydown(e) {
+      const item = e.target;
```

Comment:
It's possible to have something else than a `vaadin-context-menu-item` here.
When clicking at the edge of the overlay, the whole overlay is focused and that will be the event target. Then the page can be scrolled with arrow keys, and Up arrow throws an error.

Reply:
Fixed. At least the error is no longer thrown, I'm not sure what should we do about focusing the overlay though. It might be better to discuss in person and possibly fix later.

Reply:
Well that's a really special case, and it's not easily reproduced by accident, so I think this is good enough for now.